### PR TITLE
Add long help to osc project

### PR DESCRIPTION
**Pull request by nak3**
@deads2k
I know this PR is different from no.2 which you mentioned at [pull/1929](https://github.com/openshift/origin/pull/1929).
But listing project by osc is necessary and this PR will fix some error handling.
Can I ask you to review?

### CODE REVIEW

**Comment by deads2k** [comment]
I like the additional help, but we support listing projects via `osc get projects`.  Are you proposing behavior different from `osc get projects`?  @smarterclayton has pitched a similar idea where `osc project -l` gives you a detailed list of the projects/contexts in your kubeconfig and maybe the top few hits from the api server, but the idea hadn't been fully spec'ed.

Continuing with where you've started with the more detailed help, I do have a wishlist of similar changes:
 1. taking name and full name as arguments
 1. long help (you've got here)
 1. structured examples.  See `create_authinfo.go` in kubernetes
 1. bash completion.  Cobra has some bash completion structure for arguments if you felt really froggy.

Those items need doing across most of our commands, but I'll definitely take them on individual ones as well.

File: pkg/cmd/cli/cmd/project.go [hunk 1]
```diff
@@ -92,17 +105,31 @@ func (o ProjectOptions) Validate() error {
 }
 
 // RunProject contains all the necessary functionality for the OpenShift cli project command
-func (o ProjectOptions) RunProject() error {
+func (o ProjectOptions) RunProject(cmd *cobra.Command) error {
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/project.go]: We're trying to create a pattern where one struct holds all the options required to perform a command.  The command implementation itself builds the options up, runs a validate on them, and then calls a no-arg `run`.  That means that `projectList` that you're pulling below would more properly live in the `ProjectOptions` struct.

That said, I think we'll want to consider whether we want a different alias for `osc get projects` or something different.  See my other comment.

**Comment by smarterclayton** [comment]
I'd like to start with osc project -l that returns all of your local contexts (and the user, server if they differ).  At the bottom of that you can say "Call `osc get projects` to see the other projects you have access to on this server."

----- Original Message -----

**Comment by nak3** [comment]
Thank you for your comments. I understand. As you said, `osc project -l` in my PR is same with `osc get projects`.
I fix to add long help here.

**Comment by derekwaynecarr** [comment]
LGTM

File: pkg/cmd/cli/cli.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func NewCommandCLI(name, fullName string) *cobra.Command {
 	out := os.Stdout
 
 	cmds.AddCommand(cmd.NewCmdLogin(f, in, out))
-	cmds.AddCommand(cmd.NewCmdProject(f, out))
+	cmds.AddCommand(cmd.NewCmdProject(fullName, f, out))
```
Comment (deads2k) [inline pkg/cmd/cli/cli.go]: While you're rebasing, can you update this to look more like the line right below?  The intent is to pass the short name of the command to get the `Use` right and the `fullName` to be the complete name without any appending.  So `"project"` and `fullName + " project"`.  

Don't forget to update the long description when you make the change.

**Comment by deads2k** [comment]
One item to tidy up during the rebase, otherwise lgtm.

**Comment by nak3** [comment]
Thank you! I changed it.

**Comment by deads2k** [comment]
I still see merge conflicts on this.  Can you try rebasing again?

**Comment by nak3** [comment]
Sorry, it was my fault. I think it will be all right this time.

**Comment by deads2k** [comment]
[merge]

**Pull request merged**